diff --git a/nginx.spec b/nginx.spec
index e829ec7..0fce5f0 100644
--- a/nginx.spec
+++ b/nginx.spec
@@ -44,10 +44,10 @@ Requires(pre): pwdutils
 
 # end of distribution specific definitions
 
-Summary: High performance web server
+Summary: High performance web server (built with Google PageSpeed module)
 Name: nginx
 Version: 1.6.2
-Release: 1%{?dist}.ngx
+Release: 2%{?dist}.ngx
 Vendor: nginx inc.
 URL: http://nginx.org/
 
@@ -61,6 +61,7 @@ Source6: nginx.vh.example_ssl.conf
 Source7: nginx.suse.init
 Source8: nginx.service
 Source9: nginx.upgrade.sh
+Source10: ngx_pagespeed
 
 License: 2-clause BSD-like license
 
@@ -83,6 +84,7 @@ Not stripped version of nginx built with the debugging log support.
 
 %prep
 %setup -q
+cp -R -p %SOURCE10 .
 
 %build
 ./configure \
@@ -120,6 +122,7 @@ Not stripped version of nginx built with the debugging log support.
         --with-debug \
         %{?with_spdy:--with-http_spdy_module} \
         --with-cc-opt="%{optflags} $(pcre-config --cflags)" \
+        --add-module=%{_builddir}/%{name}-%{version}/ngx_pagespeed \
         $*
 make %{?_smp_mflags}
 %{__mv} %{_builddir}/%{name}-%{version}/objs/nginx \
@@ -158,6 +161,7 @@ make %{?_smp_mflags}
         --with-ipv6 \
         %{?with_spdy:--with-http_spdy_module} \
         --with-cc-opt="%{optflags} $(pcre-config --cflags)" \
+        --add-module=%{_builddir}/%{name}-%{version}/ngx_pagespeed \
         $*
 make %{?_smp_mflags}
 
