diff --git a/nginx.spec b/nginx.spec
index 53b3101..e2c1e03 100644
--- a/nginx.spec
+++ b/nginx.spec
@@ -56,10 +56,10 @@ Requires: systemd
 
 # end of distribution specific definitions
 
-Summary: High performance web server
+Summary: High performance web server (built with Google PageSpeed module)
 Name: nginx
 Version: 1.7.12
-Release: 1%{?dist}.ngx
+Release: 2%{?dist}.ngx
 Vendor: nginx inc.
 URL: http://nginx.org/
 
@@ -74,6 +74,7 @@ Source7: nginx.suse.init
 Source8: nginx.service
 Source9: nginx.upgrade.sh
 Source10: nginx.suse.logrotate
+Source11: ngx_pagespeed
 
 License: 2-clause BSD-like license
 
@@ -100,6 +101,7 @@ Not stripped version of nginx built with the debugging log support.
 
 %prep
 %setup -q
+cp -R -p %SOURCE11 .
 
 %build
 ./configure \
@@ -137,6 +139,7 @@ Not stripped version of nginx built with the debugging log support.
         --with-debug \
         %{?with_spdy:--with-http_spdy_module} \
         --with-cc-opt="%{optflags} $(pcre-config --cflags)" \
+	--add-module=%{_builddir}/%{name}-%{version}/ngx_pagespeed \
         $*
 make %{?_smp_mflags}
 %{__mv} %{_builddir}/%{name}-%{version}/objs/nginx \
@@ -175,6 +178,7 @@ make %{?_smp_mflags}
         --with-ipv6 \
         %{?with_spdy:--with-http_spdy_module} \
         --with-cc-opt="%{optflags} $(pcre-config --cflags)" \
+	--add-module=%{_builddir}/%{name}-%{version}/ngx_pagespeed \
         $*
 make %{?_smp_mflags}
 
